### DISCUSSION TITLE
Build issue (HIPBLAS for gfx600)

### DISCUSSION THREAD

**Discussion by danilabagroff** [ggml-org/llama.cpp#9433] [Q&A] [answered]
Hi folks,

I am trying to build llama.cpp for my AMD FirePro D500 with `-DGGML_HIPBLAS=ON  -DAMDGPU_TARGETS=gfx600` but stuck at the very beginning:
`/opt/rocm-6.2.0/lib/llvm/bin/../../../include/hip/hip_runtime.h:41:2: error: HIP is not supported on the specified GPU ARCH with wavefront size 64`

I have find out(https://clang.llvm.org/docs/AMDGPUSupport.html) that we have a macro:
`__AMDGCN_WAVEFRONT_SIZE__ | Defines the wavefront size. Allowed values are 32 and 64.
__AMDGCN_WAVEFRONT_SIZE | Alias to __AMDGCN_WAVEFRONT_SIZE__. To be deprecated.`

I have tried to define __AMDGCN_WAVEFRONT_SIZE=32 within DCMAKE_CXX_FLAGS without any luck :(
Do I have any chance to build llama.cpp for my target GPU?

Thank you.

**Comment by 8XXD8** [comment, marked as answer]
`gfx600` is unsupported, and has been for years. Rocm 6.2 officially only supports `gfx906` (Mi50 and Radeon VII) an up, you can make `gfx900` (Mi25 and Vega) work but that is all, anything older is a no go sadly.
Maybe the `Vulkan` backend can work.

**Reply by Allan-Luu** [reply]
maybe an older version of ROCm would work? try using a docker container that contains maybe `ROCm 5.7` or `ROCm 6.0` if `ROCm 6.2` doesn't support `gfx600`

**Comment by FlorianHeigl** [comment]
I heard that debian is running their own builders enabling missing AMD platforms. If you got the opportunity, you could try that. so debian os and amdgpu, rocm etc from the _DEBIAN_ repos. _NOT from AMD.
